src/abstractitensornetwork.jl

```diff
@@ -721,6 +721,48 @@ function contract_inner(
   return contract(tn; sequence)[]
 end
 
+function contract_with_BP(
+  itn::AbstractITensorNetwork; outputlevel=1, partitioning=group(v -> v, vertices(itn))
+)
+  return contract_with_BP(
+    ComplexF64, itn::AbstractITensorNetwork; outputlevel, partitioning
+  )
+end
+
+function contract_with_BP(
```

Comment:
`contract_with_BP` isn't a good function name, what about overloading `ITensors.contract(::Algorithm"bp", kwargs...)` and running with `contract(itn; alg="bp", kwargs...)`.